server/console_storage.go

```diff
@@ -111,17 +123,83 @@ func (s *ConsoleServer) ListStorage(ctx context.Context, in *console.ListStorage
 		}
 		userID = &uid
 	}
-
 	var query string
+
+	collections := make([]string, 0)
+	query = "SELECT DISTINCT collection FROM storage"
```

Comment:
I don't think we have an index on only `collection` - so we'll likely do a full index scan? Have you run a EXPLAIN statement on this?

Reply:
@michal-harish This will need to be cached on the server. The query is expensive but we also don't want an additional index because it introduces unwanted write amplification around the storage engine. Best to cache and refresh the values used in the dropdown periodically. Let me know if you want to talk this through.

Reply:
ok what we'll do is:
- create an in-memory cache set which will be nil
- when a write comes and the cache is nil do nothing
- when a call to /v2/storage-collections occurs and the cache is nil we'll do a full scan an initialize the cache (write-lock)
- when a storage object write comes and the cache is not-nil we'll update the entry for its collection value (write-lock)
- when a call to /v2/storage-collections occurs and the cache is not-nil we'll return the cache content (read-lock)